app/code/Magento/CatalogGraphQl/Model/Resolver/Products/DataProvider/Image/Placeholder.php

```diff
@@ -52,8 +71,12 @@ public function getPlaceholder(string $imageType): string
             return $imageAsset->getUrl();
         }
 
-        return $this->assetRepository->getUrl(
-            "Magento_Catalog::images/product/placeholder/{$imageType}.jpg"
-        );
+        $storeId = $this->storeManager->getStore()->getId();
+
+        $this->appEmulation->startEnvironmentEmulation($storeId, Area::AREA_FRONTEND, true);
+        $url = $this->assetRepository->getUrl("Magento_Catalog::images/product/placeholder/{$imageType}.jpg");
+        $this->appEmulation->stopEnvironmentEmulation();
```

Comment:
I really don't like using `appEmulation` as it causes more side effects than benefits.
I will approve that right now, but later on, we need to find a better solution than wrapping it this way.

Reply:
Sorry, it should not be approved

@lbajsarowicz is right. It will be lead to huge performance degradation.